src/ocaml_lsp.mli

```diff
@@ -4,7 +4,19 @@ type t
 
 val of_initialize_result : LanguageClient.InitializeResult.t -> t
 
-val has_interface_specific_lang_id : t -> bool
+val is_version_up_to_date :
+     t
+  -> Ocaml_version.t
+  -> ( bool
+     , [> `Ocaml_version_not_supported of Ocaml_version.t
+       | `Unexpected of
+         [> `Language_server_isn't_ocamllsp
+         | `Missing_serverInfo
+         | `Unable_to_parse_version
+         | `ServerInfo_version_missing
```

Comment:
Some of these error constructors are only used for error construction. In these cases, we shouldn't even have these constructors as all they accomplish is leaking implementation information. I haven't looked closely but I'm pretty sure at least the last 3 constructors can be eliminated into `Message of string`.